modules/presets/collection.js

```diff
@@ -58,26 +58,48 @@ export function presetCollection(collection) {
                 return index === 0;
             }
 
-            function sortNames(a, b) {
-                var aCompare = (a.suggestion ? a.originalName : a.name()).toLowerCase();
-                var bCompare = (b.suggestion ? b.originalName : b.name()).toLowerCase();
+            function getMatchWord(match) {
+                if (match.alias) return match.alias;
+                if (match.term) return match.term;
+                if (match.tagValue) return match.tagValue;
+                if (match.preset.suggestion) return match.preset.originalName;
+                return match.preset.name();
+            }
+
+            function sortMatches(a, b) {
+                var aCompare = getMatchWord(a).toLowerCase();
+                var bCompare = getMatchWord(b).toLowerCase();
 
-                // priority if search string matches preset name exactly - #4325
+                // priority if search string matches word exactly - #4325
                 if (value === aCompare) return -1;
                 if (value === bCompare) return 1;
 
+                // priority for smaller edit distance
+                var i = a.fuzziness - b.fuzziness;
+                if (i !== 0) return i;
+
                 // priority for higher matchScore
-                var i = b.originalScore - a.originalScore;
+                i = b.preset.originalScore - a.preset.originalScore;
                 if (i !== 0) return i;
 
-                // priority if search string appears earlier in preset name
+                // priority if search string appears earlier in word
                 i = aCompare.indexOf(value) - bCompare.indexOf(value);
                 if (i !== 0) return i;
 
-                // priority for shorter preset names
+                // priority for shorter words
                 return aCompare.length - bCompare.length;
             }
 
+            // take those presets from the source array for which the given function
+            // returns a match and sort the resulting list of matches
+            function take(array, fn) {
+                return utilArrayMapTruthy(array, function (val, i) {
+                    var r = fn.call(null, val, i, array);
+                    if (r) array[i] = null;
+                    return r;
+                }).sort(sortMatches);
+            }
+
```

Comment:
Besides making this `filter + map + sort` more concise in the code below, this is also a performance improvement: Presets that are added to the results are not searched again.